pkg/ccl/backupccl/backupresolver/targets.go

```diff
@@ -258,11 +259,16 @@ func DescriptorsMatchingTargets(
 
 	alreadyRequestedDBs := make(map[descpb.ID]struct{})
 	alreadyExpandedDBs := make(map[descpb.ID]struct{})
+	invalidRestoreTsErr := errors.Errorf("supplied backups do not cover requested time")
 	// Process all the DATABASE requests.
 	for _, d := range targets.Databases {
 		dbID, ok := resolver.DbsByName[string(d)]
 		if !ok {
-			return ret, errors.Errorf("unknown database %q", d)
+			// given a non-empty aost, it's possible that we're unable to find the database due to aost being set before gc ttl
+			if asOf.IsEmpty() {
+				return ret, errors.Errorf("unknown database %q", d)
```

Comment:
nit: can we change this and below to say - "database ... does not exist"

Reply:
done